covalent_indexer.go

```diff
@@ -187,28 +187,33 @@ func (ci *covalentIndexer) SaveBlock(args *indexer.ArgsSaveBlockData) {
 	}
 
 	ci.sendWithRetrial(dataToSend, blockResult.Block.Hash)
+
+	// TODO next PRs - remove the retrial, it is done by the node
```

Comment:
What kind of retrial are we talking about?

Do you mean that if, for instance:
1. You fail to send the message (`dataToSend`) for whatever reason(receiver hasn't sent back acknowledge signal)
2. Your return the error from `SaveBlock`
3. Node tries to call `SaveBlock` again until error is gone?

So instead of `sendWithRetrial` we will have `sendData`: 

```go
func (ci *covalentIndexer) sendData(data []byte, ackData []byte) error{
	wss := ci.getWSS()
	wsr := ci.getWSR()
	if wss == nil {
		ci.waitForWSSConnection()
	}
	if wsr == nil {
		ci.waitForWSRConnection()
	}

	return ci.sendDataWithAcknowledge(data, ackData, wss, wsr)
}
```

Reply:
yes, something like this. Will be simplified